Kudu.Core/SiteExtensions/SiteExtensionManager.cs

```diff
@@ -456,12 +456,33 @@ private SiteExtensionInfo GetPreInstalledExtension(string id)
                     {
                         extractPath = _environment.WebRootPath;
                         FileSystemHelpers.EnsureDirectory(extractPath);
-
-                        tracer.Trace("Generate xdt file to map application to run from wwwroot when request for /{0}", package.Identity.Id);
-                        GenerateRemapToWebRootApplicationHostXdt(installationDirectory, string.Format(CultureInfo.InvariantCulture, "/{0}", package.Identity.Id));
                     }
 
                     await remoteRepo.DownloadPackageToFolder(package.Identity, extractPath, pathToLocalCopyOfNudpk: packageLocalFilePath);
+
+                    if (SiteExtensionInfo.SiteExtensionType.WebRoot == type)
+                    {
+                        // if install to WebRoot, check if there is any xdf file come with package
+                        // if there is one move it to site extension folder
+                        // if not, generate one to map request to wwwroot
+                        var xdtFiles = FileSystemHelpers.GetFiles(extractPath, "*.xdt");
```

Comment:
We really only support `applicationHost.xdt`, and no others. So may as well make code specific to that, and not need any looping. You can use the `_applicationHostFile` constant.